testing/trino-testing/src/main/java/io/trino/testing/LocalSpoolingManager.java

```diff
@@ -167,26 +173,78 @@ public SpoolingManager create(Map<String, String> config, SpoolingManagerContext
         }
     }
 
-    public record LocalSpooledSegmentHandle(@Override String encoding, @Override QueryId queryId, Path path)
+    public static class LocalSpooledSegmentHandle
             implements SpooledSegmentHandle
     {
-        public LocalSpooledSegmentHandle
+        private final String encoding;
+        private final QueryId queryId;
+        private final Path path;
+
+        public LocalSpooledSegmentHandle(String encoding, QueryId queryId, Path path)
         {
-            requireNonNull(encoding, "encoding is null");
-            requireNonNull(queryId, "queryId is null");
-            requireNonNull(path, "path is null");
+            this.encoding = requireNonNull(encoding, "encoding is null");
+            this.queryId = requireNonNull(queryId, "queryId is null");
+            this.path = requireNonNull(path, "path is null");
         }
 
+        @JsonIgnore
         @Override
         public Instant expirationTime()
         {
             return Instant.MAX;
         }
 
+        @JsonProperty
+        @Override
+        public QueryId queryId()
+        {
+            return queryId;
+        }
+
+        @JsonIgnore
         @Override
         public String identifier()
         {
             return path.getFileName().toString();
         }
+
+        @JsonProperty
+        @Override
+        public String encoding()
+        {
+            return encoding;
+        }
+
+        @JsonIgnore
+        public Path getPath()
+        {
+            return path;
+        }
+
+        @JsonProperty("path")
+        public String getFilePath()
+        {
+            return path.toAbsolutePath().toString();
```

Comment:
why not keep it a record an just keep absolute path on field?

Reply:
Path is serialized differently (like an URI with a file:/// scheme)